app/src/main/kotlin/com/adesso/movee/scene/persondetail/PersonDetailViewModel.kt

```diff
@@ -25,13 +25,11 @@ class PersonDetailViewModel @Inject constructor(
 
     fun fetchPersonDetails(personId: Long) {
         if (_personDetails.value == null) {
-            bgScope.launch {
-                val personDetailResult =
-                    fetchPersonDetailsUseCase.run(FetchPersonDetailsUseCase.Params(personId))
-
-                onUIThread {
-                    personDetailResult.either(::handleFailure, ::postPersonDetails)
-                }
+            fetchPersonDetailsUseCase.invoke(
+                viewModelScope,
+                FetchPersonDetailsUseCase.Params(personId)
+            ) {
+                it.either(::handleFailure, ::postPersonDetails)
```

Comment:
postPersonDetails() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postPersonDetails()